### ISSUE TITLE
minitest testsuite failure with ruby3.4.0dev

### ISSUE THREAD

**Issue by mtasaka** [jekyll/jekyll#9721]
Testing with `ruby 3.4.0dev (2024-11-29 master 84b14c0385) +PRISM [x86_64-linux]` 
https://github.com/ruby/ruby/commit/84b14c03851d919a99766386c619fe703134c36c

On jekyll git head https://github.com/jekyll/jekyll/commit/0e4182aefad27c72c6b1c0f0e300e57edefaa0ba
minitest testsuite fails like:

```
+ env TZ=UTC ruby -Ilib:test -e 'Dir.glob "./test/**/test_*.rb", &method(:require)'
# -------------------------------------------------------------
# SPECS AND TESTS ARE RUNNING WITH WARNINGS OFF.
# SEE: https://github.com/Shopify/liquid/issues/730
# SEE: https://github.com/jekyll/jekyll/issues/4719
# -------------------------------------------------------------
Run options: --seed 43620


...

Finished in 35.562540s, 25.3919 runs/s, 56.4920 assertions/s.

  1) Failure:
TestFilters#test_: filters inspect filter should return a HTML-escaped string representation of an object.  [test/test_filters.rb:1460]:
--- expected
+++ actual
@@ -1 +1 @@
-"{&quot;&lt;a&gt;&quot;=&gt;1}"
+"{&quot;&lt;a&gt;&quot; =&gt; 1}"


903 runs, 2009 assertions, 1 failures, 0 errors, 0 skips
```

This is most likely due to 
https://github.com/ruby/ruby/blob/f9d0bc22f5ca019f3c517b42ea4a50867fe56700/NEWS.md?plain=1#L240-L243
https://bugs.ruby-lang.org/issues/20433

**Comment by ashmaroli** [comment]
Thank you for reporting this @mtasaka.
However, we currently do not have plans to run our CI on `ruby-head` and will therefore wait for the upcoming release and subsequent complaints from our users.

You are welcome to submit a pull request to update our CI workflow (with necessary fixes to lib / test files) once Ruby 3.4 releases. I will give your pull request submission higher priority over another that makes the same proposal in case there is a delay in you submitting a pull request owing to the festive season.

That said, I feel this particular test failure is going to need a clunky fix (like `result = RUBY_VERSION <= "3.4" ? something : something_else`). If you can propose a better fix, all the more welcome.

**Comment by ashmaroli** [comment]
Hello @mtasaka, just checking if you are planning on a pull request towards this. If not, I shall patch this myself.
